server/mqtt.go

```diff
@@ -1374,22 +1374,28 @@ func (s *Server) mqttCreateAccountSessionManager(acc *Account, quitCh chan struc
 		}
 	}
 
-	// Try to transfer regardless if we have already updated the stream or not
-	// in case not all messages were transferred and the server was restarted.
-	if needToTransfer {
+	transferRMS := func() {
+		if !needToTransfer {
+			return
+		}
 		err = as.transferRetainedToPerKeySubjectStream(s)
 		if err != nil {
-			return nil, err
+			return
 		}
-
 		// We need another lookup to have up-to-date si.State values in order
 		// to load all retained messages.
 		si, err = lookupStream(mqttRetainedMsgsStreamName, "retained messages")
 		if err != nil {
-			return nil, err
+			return
```

Comment:
Well, for this, a failure is bad because it means that `si`, which is expected to reflect the latest state, will be not the correct one.

Reply:
Right. Thx!

Reply:
should be resolved now - ignore the first time (since will retry anyway), fail the connection on the 2nd, if errors.